app/controllers/likes_controller.rb

```diff
@@ -1,45 +1,44 @@
 class LikesController < ApplicationController
   before_action :authenticate_user!
+  before_action :find_like, only: [:destroy]
+
 
 
-=begin
   def create
+    session[:return_to] ||= request.referer
     @post = Post.find(params[:post_id])
+    @post.likes.create(user: current_user, post: @post)
+    @like = @post.likes.where(user: current_user, post: @post)[0]
     #debugger
-    @like = @post.likes.create(like_params) do |like|
-      like.user_id=current_user[:id]
-    end
-    #debugger
-    if @like.save
-      redirect_to root_path
+    respond_to do |format|
+      format.js
+      #format.html { redirect_to posts_url, notice: 'Post was successfully destroyed.' }
+      #format.json { head :no_content }
     end
 
+
+
   end
-=end
 
-  def create
-    session[:return_to] ||= request.referer
-    @post = Post.find(params[:post_id])
-    if @post.likes.where(user: current_user.id).present?
-        flash[:error] = "You've already upvoted this!"
-    elsif @post.likes.create(user: current_user, post: @post)
-        flash[:success] = "Upvoted!"
-    else
-        flash[:error] = "An error prevented you from upvoting."
-    end
+  def destroy
+    #session[:return_to] ||= request.referer
 
-    #respond_to do |format|
-      #format.js
+    @like.destroy
+    respond_to do |format|
+      format.js
       #format.html { redirect_to posts_url, notice: 'Post was successfully destroyed.' }
       #format.json { head :no_content }
-    #end
-    #redirect_to root_path
-    redirect_to session.delete(:return_to)
-    #render session.delete(:return_to)
+    end
+    #redirect_to session.delete(:return_to)
+    #redirect_to post_path(@post)
   end
 
   private
     def like_params
       params.permit(:post_id)
     end
+    def find_like
+      @post = Post.find(params[:post_id])
+      @like = @post.likes.find(params[:id])
```

Comment:
again you assuming for each post a like must exit. Why are not considering the other cases?
This above code will break.

Reply:
"find_like" only executes before the destroy action and destroy buttons only appears when some one liked the post so in that case to dislike a post a like must exist for that post.

Reply:
What will happen it two request execute same function. In case of race condition this function will break for second one. 
Please handle all cases.

Reply:
Done